gunicorn/arbiter.py

```diff
@@ -495,6 +495,17 @@ def manage_workers(self):
             (pid, _) = workers.pop(0)
             self.kill_worker(pid, signal.SIGTERM)
 
+        if self.pidfile is None and self.cfg.pidfile is not None and self.cfg.delay_pidfile:
+            worker_values = list(self.WORKERS.values())
+            booted = True
+            for worker in worker_values:
+                if not worker.booted:
```

Comment:
nice!

Given that self.wsgi = self.app.wsgi() is called before self.booted is set to True https://github.com/lyft/gunicorn/blob/delay-pid-creation/gunicorn/workers/base.py#L114-L119 I assume that getting the wsgi app will import the whole app and bootstrap everything.

It's definitely worth a try. I'd actually document the reasoning (^^) in a comment.